promadapter/conversion.go

```diff
@@ -5,22 +5,38 @@ import (
 
 	"time"
 
+	"github.com/appoptics/appoptics-api-go"
 	"github.com/prometheus/common/model"
 	promremote "github.com/prometheus/prometheus/storage/remote"
-	"github.com/appoptics/appoptics-api-go"
 )
 
 //
 // The adapter package holds information necessary to convert from Prometheus types to types defined in the AppOptics
 // client library, as well as for creating API-compliant batches and using the AppOptics client to send them.
 //
 
-func PromDataToAppOpticsMeasurements(req *promremote.WriteRequest) []appoptics.Measurement {
-	return SamplesToMeasurements(WriteRequestToSamples(req))
+type PrometheusAdapter interface {
+	WriteRequestToSamples(req *promremote.WriteRequest) model.Samples
+	PromDataToAppOpticsMeasurements(req *promremote.WriteRequest) []appoptics.Measurement
+	SamplesToMeasurements(samples model.Samples) []appoptics.Measurement
+	LabelsToTags(sample *model.Sample) map[string]string
+}
+
+type Adapter struct {
```

Comment:
warning: exported method Adapter.PromDataToAppOpticsMeasurements should have comment or be unexported (golint)